File: api/array.go
package handler

import (
	"fmt"
	"net/http"
	"time"
)

func Handler(w http.ResponseWriter, r *http.Request) {
	s := []string{}
	start := time.Now()
	for i := 0; i < 5000; i++ {
		s = append(s, "test")
	}
	duration := time.Since(start)
	// fmt.Fprintf(w, "len=%d cap=%d %v\n", len(s), cap(s), s)
	fmt.Fprintf(w, "%f ms", float64(duration.Nanoseconds())/1000000)
}

File: api/rust/array.rs
use std::time::Instant;
use vercel_runtime::{run, Body, Error, Request, Response, StatusCode};

#[tokio::main]
async fn main() -> Result<(), Error> {
    run(handler).await
}

pub async fn handler(_req: Request) -> Result<Response<Body>, Error> {
    let mut v = Vec::new();
    let start = Instant::now();
    for _ in 0..1000000 {
        v.push("test");
    }
    let duration = start.elapsed();
    let time = format!("{:?} ms", duration.as_secs_f64() * 1000.0);
    Ok(Response::builder()
        .status(StatusCode::OK)
        .body(time.into())?)
}
